ext/objspace/objspace_dump.c

```diff
@@ -542,7 +547,7 @@ dump_object(VALUE obj, struct dump_config *dc)
 
       case T_OBJECT:
         dump_append(dc, ", \"ivars\":");
-        dump_append_lu(dc, ROBJECT_IV_CAPACITY(obj));
+        dump_append_lu(dc, ROBJECT_IV_COUNT(obj));
```

Comment:
Do you think we could add a properly to show which objects are using that special shape? I assume they'll all have `shape_id: OBJ_TOO_COMPLEX_SHAPE_ID`, and that `OBJ_TOO_COMPLEX_SHAPE_ID` should be `1`? But maybe making it a bit more explicit would be helpful?

Or maybe looking for `variation_count: 8` in the dump is enough? (mostly thinking out loud here).

Reply:
The "too complex" shape id is 11 right now, but that could change in the future.  We currently [dump the name of every shape](https://github.com/jemmaissroff/ruby/blob/af586c10f4645a032e23c86c9ac7aae45a81f427/ext/objspace/objspace_dump.c#L766-L767), and every object also [shows its shape id](https://github.com/jemmaissroff/ruby/blob/af586c10f4645a032e23c86c9ac7aae45a81f427/ext/objspace/objspace_dump.c#L408-L412), so it's possible to check the shape id of the object, then check the shape name and be able to understand if the object is too complex.

I'm not against dumping a boolean value for "too complex" though as that's probably a common thing people will want to know.

Reply:
> that's probably a common thing people will want to know.

Yup.

I don't want to block this PR for that though, so let's merge this one, and I can open another PR to refine the `ObjectSpace.dump` part of it tomorrow.